src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabSCMSource.java

```diff
@@ -304,7 +304,7 @@ protected void retrieve(SCMSourceCriteria criteria, @NonNull SCMHeadObserver obs
         try {
             GitLabApi gitLabApi = apiBuilder(serverName);
             getGitlabProject(gitLabApi);
-            setProjectId(gitlabProject.getId());
+            setProjectId(gitlabProject.getId().toString());
```

Comment:
```suggestion
            setProjectId(gitlabProject.getId());
```

Reply:
Why not pass the already built gitlabapi? 😕 and use the result from getGitlabProject? 🤔